tests/net/socket/misc/src/main.c

```diff
@@ -51,6 +51,24 @@ ZTEST_USER(socket_misc_test_suite, test_inet_pton)
 
 	res = zsock_inet_pton(AF_INET6, "a:b:c:d:0:1:2:3z", buf);
 	zassert_equal(res, 0, "");
+
+	res = zsock_inet_pton(AF_INET, "192.0.2.400", buf);
+	zassert_equal(res, 0, "");
+
+	res = zsock_inet_pton(AF_INET, "0.0", buf);
```

Comment:
I might have missed some discussion, but I'm not sure I follow why should these be considered a valid IP addresses? Even checked with native Linux app, and they're all rejected by `inet_pton()`:
```
inet_pton returned 0 for 0.0
inet_pton returned 0 for 0.1
inet_pton returned 0 for 0
inet_pton returned 0 for 1
```

Reply:
I think they probably are not valid IP addresses.

Reply:
Ok, lets ban these short addresses then. I have seen these used as IP addresses but it probably was not with `inet_pton` but with some other tool.

Reply:
I do not think they are invalid at all, they may be unusable. Those adresses may be invalid on the public internet or have a designated not "free for all" usage.
The IANA list: https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml for example lists addresses 0.0.0.0 and its subnet as valid but to limited use.